README.md

```diff
@@ -1,3 +1,95 @@
-# storey
+# Storey
 
 [![CI](https://github.com/v3io/storey/workflows/CI/badge.svg)](https://github.com/v3io/storey/actions?query=workflow%3ACI)
+
+Storey is an asynchronous flow library, for real time event processing and feature extraction.
+
+#### In This Document
+
+- [API Walkthrough](#api-walkthrough)
+- [Usage Examples](#examples)
+
+
+<a id="api-walkthrough"></a>
+## API Walkthrough
+A Storey flow consist of steps linked together by the `build_flow` function, each doing it's designated work.
+
+### Supported Steps
+#### Input Steps
+* `Source` - 
+* `AsyncSource` - 
+* `ReadCSV` - 
+
+#### Processing Steps
+* `Filter` - 
+* `Map` -  
+* `FlatMap` -
+* `MapWithState` -
+* `Batch(max_events, timeout)` - Batches multiple events and emits one event, containing all the batch data, when reaching `max_events` or `timeout` has passed since the first event in the batch.
```

Comment:
Batches events. This step emits a batch every `max_events` events, or when `timeout` seconds have passed since the first event in the batch was received.